context/state.tsx

```diff
@@ -162,28 +123,28 @@ const saveState = (state: tezosState) => {
   );
 };
 
-function reducer(state: tezosState, action: action): tezosState {
+const loadState = (userAddress: string) => {
+  const rawStorage = localStorage.getItem(`app_state:${userAddress}`);
+  if (!rawStorage) return {};
+  return JSON.parse(rawStorage);
+};
+
+function reducer(
+  state: tezosState,
+  action: action,
+  { userAddress }: { userAddress: string }
+): tezosState {
   switch (action.type) {
-    case "beaconConnect": {
-      state.connection.setProvider({
-        rpc: RPC_URL,
-        wallet: action.payload,
-      });
-      state.connection.setSignerProvider(new BeaconSigner(action.payload));
-      return { ...state, beaconWallet: action.payload };
-    }
     case "p2pConnect": {
       return { ...state, p2pClient: action.payload };
     }
     case "addDapp": {
-      if (!state.address) return state;
```

Comment:
The address is always defined? I don't know what is `addDapp` but can it be triggered when the user is disconnected without any safe?

Reply:
address is no longer defined in the global state but in wallet context.
`addDapp` allows to connect an external dapp with Tzsafe and user have to import a contract to access to this feature, so I believe disconnected user cannot trigger this action